pkg/destroy/gcp/cloudcontroller.go

```diff
@@ -3,42 +3,74 @@ package gcp
 import (
 	"context"
 	"fmt"
+	"regexp"
+	"strings"
 
-	compute "google.golang.org/api/compute/v1"
+	"google.golang.org/api/compute/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
+type resourceFilterFunc func(string) bool
+
+func (o *ClusterUninstaller) hasClusterIDPrefix(itemName string) bool {
+	return o.isClusterResource(itemName)
+}
```

Comment:
We can consolidate this into a single function, either `hasClusterIDPrefix` or `isClusterResource`